### PR TITLE
n3h mode default to REAL

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary

As discussed with @ddd-mtl, the n3h mode to use as default should be `REAL` now.

## changelog

Please check one of the following, relating to the [CHANGELOG](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG.md)

- [x] this is a code change that effects some consumer (e.g. zome developers) of holochain core so there is an 'Unreleased' CHANGELOG item, with the format `- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)`
- [ ] this is not a code change, or doesn't effect anyone outside holochain core development

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Increased timeout on n3h spawn and wait for `#P2P-READY#` message [#1276](https://github.com/holochain/holochain-rust/pull/1276).
+- Default N3H mode as set when spawned by the conductor got set to "REAL". [#1282](https://github.com/holochain/holochain-rust/pull/1282)
```
Comment (Connoropolous) [inline CHANGELOG.md]: Could we take this opportunity to clarify internally and externally what those nodes actually are? I personally don’t even know, in clear terms

Comment (ddd-mtl) [reply CHANGELOG.md]: REAL is a big update to HACK mode.
HACK is going to dissapear shortly. 
Consider REAL to be HACK mode's new name.
I needed to keep both around to make sure things didn't break.

REAL is in opposition to MOCK mode.
Mock mode is where you connect multiple local nodes to one n3h in order to simulate a network over IPC.
REAL is the normal n3h mode that gossips over the network.
Maybe renamed it to NORMAL?

Is that clear enough for you?

Comment (Connoropolous) [reply CHANGELOG.md]: Yep

Comment (thedavidmeister) [reply CHANGELOG.md]: @Connoropolous @ddd-mtl can it be a small followup to do a rename and document this?

File: conductor_api/src/config.rs [hunk 2]
```diff
@@ -507,7 +507,16 @@ pub struct NetworkConfig {
...
+    /// Overall mode n3h operates in.
+    /// Should be one of
+    /// * REAL
+    /// * MOCK
+    /// * HACK
+    /// REAL is the default and what should be used in all production cases.
+    /// MOCK is for using n3h only as a local hub that apps connect to directly, i.e. n3h will
+    /// not connect to any other n3h instance.
+    /// HACK is for trying out things and should only be used by n3h developers who know why
+    /// they need this (i.e. testing new code). Might get removed soon.
```
Comment (lucksus) [inline conductor_api/src/config.rs]: @ddd-mtl, is this correct?
@Connoropolous, is this sufficient?

Comment (ddd-mtl) [reply conductor_api/src/config.rs]: Yes, except for HACK. I'd write:
```
/// HACK - Deprecated. Used by n3h developers only. Will get removed soon.
```

**Comment by lucksus** [comment]
@thedavidmeister, @Connoropolous, @ddd-mtl, ok to merge?

**Comment by lucksus** [comment]
I‘ve turned this into a WIP since REAL Mode seems to have problems still while HACK works. Let us merge this _after_ REAL mode got fixed.

**Comment by ddd-mtl** [comment]
@lucksus Agree on postponing this. 
On my side, waiting on Core to do integration test and isolating the hypothetical problem before looking into a potential fix.

**Comment by lucksus** [comment]
@ddd-mtl, did we not identify conceptual issues that will take ~ 2 weeks solving anyway? Shouldn't we start working on that asap? @neonphog?

**Comment by lucksus** [comment]
Also, until we have core <> n3h integration tests we can only manually run apps to test integration as I did so far with holochain-basic-chat. But that is a possibility in the now. I stopped doing that for solving the issues in REAL mode because HACK mode seems to be working and sufficient for getting to working apps via the internet as soon as we solve the IPv6/zerotier issues..

**Comment by ddd-mtl** [comment]
@lucksus Yes, the other issue is n3h stores entries & metas without waiting for Core to validate it. This is not blocking so not a priority.

**Pull request merged**